docs/components/pagination.md

```diff
@@ -11,275 +11,298 @@
+const currPage = ref<number>(2)
+</script>
+```
 
-export default defineComponent({
-  setup() {
-    const currPage = ref(2)
+### offset
 
-    return {
-      currPage,
-    }
-  }
-})
+Set this prop to `true` if you want to use offset pagination. Defaults to `false`.
+
+Offset pagination is a simpler version of pagination that omits the concept of pages. When using this type of pagination you will have to keep track of current offset and related logic in your component.
+
+<div>
+  <span><b>Visible letters: </b></span>
+  <span v-for="letter in offsetVisibleLetters">{{ letter }}</span>
+  <KPagination
+    offset
+    :offset-previous-button-disabled="currentOffset === 0"
+    :offset-next-button-disabled="(currentOffset + 1) === (offsetLetters.length/offsetPageSize)"
+    :items="offsetLetters"
+    :total-count="offsetLetters.length"
+    :page-sizes="[offsetPageSize]"
+    @get-next-offset="getNextOffset"
+    @get-previous-offset="getPreviousOffset" />
+</div>
+
+```vue
+<template>
+  <span><b>Visible letters: </b></span>
+  <span v-for="letter in visibleLetters">{{ letter }}</span>
+  <KPagination
+    offset
+    :offset-previous-button-disabled="currentOffset === 0"
+    :offset-next-button-disabled="(currentOffset + 1) === (letters.length/pageSize)"
+    :items="letters"
+    :total-count="letters.length"
+    :page-sizes="[pageSize]"
+    @get-next-offset="getNextOffset"
+    @get-previous-offset="getPreviousOffset" />
+</template>
+
+<script setup lang="ts">
+const pageSize = 3
+const letters = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i']
+const currentOffset = ref<number>(0)
+const visibleLetters = ref<string>(offsetLetters.slice((currentOffset.value * 3), ((currentOffset.value * 3) + pageSize)))
+
+const getNextOffset = () => {
+  currentOffset.value++
+  visibleLetters.value = offsetLetters.slice((currentOffset.value * 3), ((currentOffset.value * 3) + pageSize))
+}
+
+const getPreviousOffset = () => {
+  currentOffset.value--
+  visibleLetters.value = offsetLetters.slice((currentOffset.value * 3), ((currentOffset.value * 3) + pageSize))
+}
 </script>
 ```
 
-### paginationType
+### offsetPreviousButtonDisabled
 
-Pass in the type of pagination to be used. Options are `default` (page/pageSize) or `offset` (offset/pageSize)
+Pass in a boolean value for whether or not the offset-based Previous button should be disabled.
 
-### offsetPrevButtonDisabled
+### offsetNextButtonDisabled
 
-Pass in a boolean value for whether or not the offset-based Previous button should be disabled
+Pass in a boolean value for whether or not the offset-based Next button should be disabled.
 
-### offsetNextButtonDisabled
+## Events
 
-Pass in a boolean value for whether or not the offset-based Next button should be disabled
+### pageChange
 
-## Usage
+Emitted when the page has been changed. The event provides following information. All of the numbers are sent in a human-understandable base (starting at 1 instead of 0) to be used directly in UI.
```

Comment:
```suggestion
Emitted when the page has been changed. The page number is returned starting at 1 rather than 0 to be easier to use directly in the UI.
```